File: Linked_List/148.Sort_List/Sort_List.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* sortList(ListNode* head) {
        vector<int> arr;
        ListNode* ans = head;
        ListNode* root = head;
        while(root){
            arr.push_back(root->val);
            root = root->next;
        }
        
        sort(arr.begin(), arr.end());
        
        int curr = 0;
        
        while(ans){
            ans->val = arr[curr];
            curr++;
            ans = ans->next;
        }
        
        return head;
        
    }
};

File: Linked_List/148.Sort_List/Sort_List.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def sortList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        root = head
        ans = head
        arr = []
        while(root):
            arr.append(root.val)
            root = root.next
        
        arr.sort()
        curr = 0
        
        while(ans):
            ans.val = arr[curr]
            ans = ans.next
            curr += 1
        
        
        return head
        